Assets/Scripts/InteractLogic.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Interfaces;
+using Player;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+
+[RequireComponent(typeof(IMovable))]
+public class InteractLogic : MonoBehaviour
+{
+    /// <summary>
+    /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+    /// (Won't conflict with character attacking enemy or moving on the map)
+    /// </summary>
+    /// 
+
+    [Tooltip("Showing the object player is focusing")]
+    [SerializeField]
+    private Interactable focus;
+
+    [Tooltip("Showing the object target player is interacting")]
+    [SerializeField]
+    private Transform target;
+
+    private IMovable _movable;
+
+    private NavMeshAgent _navMeshAgent;
+
+    private void Awake()
+    {
+        _movable = GetComponent<IMovable>();
+        _navMeshAgent = GetComponent<NavMeshAgent>();
+    }
+
+    private void Update()
+    {
+        // Check and makes character facing the target object when moving toward it
+        if (target != null) _movable.FaceTarget(target.gameObject, true, out _);
+    }
+    // This will make character going to target object and stop when it is close enough
+    private void SetTarget(Interactable newTarget)
+    {
+        _navMeshAgent.stoppingDistance = newTarget.Radius * .8f;
```

Comment:
Is there a reason for * .8f ?

Reply:
No actually. I just hard coded it and forgot to change back.